File: code-examples/pymongo/usage-examples/distinct.py
import pymongo

# Replace the uri string with your MongoDB deployment's connection string.
uri = "<connection string uri>"

client = pymongo.MongoClient(uri)

def run():
    try:
        # define a database and collection on which to run the method
        database = client["sample_mflix"]
        movies = database["movies"]

        distinct_values = movies.distinct("year", {
            "directors": "Barbra Streisand",
        })

        print(distinct_values)
    finally:
        client.close()

try:
    run()
except Exception as e:
    print(e)

File: code-examples/java-rs/usage-examples/distinct.java
import com.mongodb.reactivestreams.client.MongoClients;
import com.mongodb.reactivestreams.client.MongoClient;
import com.mongodb.reactivestreams.client.MongoDatabase;
import com.mongodb.reactivestreams.client.MongoCollection;
import org.bson.Document;
import com.mongodb.client.model.Filters;
import java.util.NoSuchElementException;

public class Main {

    public static void main(String[] args) {
        
        MongoClient mongoClient = MongoClients.create("<connection string uri>");
        MongoDatabase database = mongoClient.getDatabase("sample_mflix");
        MongoCollection<Document> movies = database.getCollection("movies");

        movies.distinct("year", Filters.eq("directors", "Barbra Streisand"), Integer.class)
            .subscribe(
                year -> System.out.println(year),
                throwable -> { throw new NoSuchElementException("No such element exists"); },
                () -> mongoClient.close()
            );
    }
}
